korge/src/commonMain/kotlin/com/soywiz/korge/input/SwipeComponent.kt

```diff
@@ -0,0 +1,139 @@
+package com.soywiz.korge.input
+
+import com.soywiz.korge.view.*
+import com.soywiz.korma.geom.*
+import kotlin.math.*
+
+data class SwipeInfo(
+    var dx: Double = 0.0,
+    var dy: Double = 0.0,
+    var direction: SwipeDirection
+)
+
+enum class SwipeDirection {
+    LEFT, RIGHT, TOP, BOTTOM
+}
+
+/**
+ * This methods lets you specify a [callback] to execute when swipe event is triggered.
+ * @property threshold a threshold for dx or dy after which a swipe event should be triggered.
+ * Negative [threshold] means a swipe event should be triggered only after [onUp] mouse event.
+ * [threshold] is set to -1 by default.
+ * @property direction a direction for which a swipe event should be triggered.
+ * [direction] is set to null by default, which means a swipe event should be triggered for any direction.
+ * @property callback a callback to execute
+ */
+fun <T : View> T.onSwipe(
+    threshold: kotlin.Number = -1,
```

Comment:
Not a big deal. I usually create a separate `inline fun` function when using `Number` and the non inline versions use Double to try to avoid boxings. Thought it is not a big deal here since this is probably not going to be called too often, but once per scene maybe